pandas/core/generic.py

```diff
@@ -1906,6 +1906,10 @@ def _slice(self, slobj, axis=0, kind=None):
         return result
 
     def _set_item(self, key, value):
+        if isinstance(key, str) and callable(getattr(self, key, None)):
```

Comment:
Using `string_types` causes failures in Python2 when creating columns that have unicode characters in their names. See c90aa2266137d78f4786b85ff3a5cd08ff0c39fd.